src/misc/host_locale_macos.cpp

```diff
@@ -30,127 +30,353 @@
-        French                     -> "fr"
-        Greek                      -> "gk"
-        Greek (Polytonic)          -> "gk"
-        Croatian                   -> "hr"
-        Hungarian                  -> "hu208"
-        Armenian                   -> "hy"
-        Hebrew                     -> "il"
-        Icelandic                  -> "is161"
-        Italian                    -> "it142"
-        Georgian                   -> "ka"
-        Spanish (Latin America)    -> "la"
-        Lithuanian                 -> "lt"
-        Latvian                    -> "lv"
-        Macedonian                 -> "mk"
-        Maltese                    -> "mt"
-        Dutch                      -> "nl"
-        Norwegian                  -> "no"
-        Northern Sami              -> "no", 30000
-        Polish                     -> "pl214""
-        Polish Pro                 -> "pl"
-        Portuguese                 -> "po"
-        Romanian                   -> "ro446"
-        Russian                    -> "ru"
-        Russian (Phonetic)         -> "ru"
-        Swiss                      -> "sd"
-        Slovene/Slovenian          -> "si"
-        Slovak                     -> "sk"
-        Swedish                    -> "sv"
-        Turkish Q                  -> "tr"
-        Turkish F                  -> "tr440"
-        Ukrainian                  -> "ua"
-        Uzbek                      -> "uz"
-        Vietnamese                 -> "vi"
-        Serbian                    -> "yc"
-        Serbian (Latin)            -> "yc"
-        // For some keyboard families we don't have code pages, but in the
-        // corresponding states the QWERTY layout is typically used
-        Bengali                    -> "us"
-        Burmese                    -> "us"
-        Chinese                    -> "us"
-        Gujarati                   -> "us"
-        Hindi                      -> "us"
-        Japanese                   -> "us"
-        Kannada                    -> "us"
-        Khmer                      -> "us"
-        Korean                     -> "us"
-        Malayalam                  -> "us"
-        Nepali                     -> "us"
-        Odia/Oriya                 -> "us"
-        Punjabi (Gurmukhi)         -> "us"
-        Sinhala                    -> "us"
-        Tamil                      -> "us"
-        Telugu                     -> "us"
-        Thai                       -> "us"
-        Tibetan                    -> "us"
-        Urdu                       -> "us"
-*/
 
+// ***************************************************************************
+// Detection data
+// ***************************************************************************
+
+// Mapping from Macintosh to DOS keyboard layout. Collected using:
+// 'System Settings' -> 'Keyboard' -> 'Text Input' -> 'Input Sources' settings
+// on macOS 'Sequoia' 15.1.1.
+static const std::map<int64_t, KeyboardLayoutMaybeCodepage> MacToDosKeyboard = {
```

Comment:
Super minor: suggest `std::unordered_map` for static constant/fixed maps. (write once, read many) because look ups are basically array-like `O(1)`, versus the ordered `std::map`, which does lookups in `O(log N)` where it has to search for the key in a tree using `<` operators.

Reply:
Suggest protecting the formatting of these big static objects with `// clang-format off` and ``// clang-format on` before and after.

Reply:
The `// clang-format off` was actually there, but I have removed it by mistake (`on` is still present at the end) - fixed.

OK, changed the `map` type - although it probably won't make much difference here (this is not a performance critical code at all).